### ISSUE TITLE
Fix Bootstrap on Mingw64 via MSYS2

### ISSUE THREAD

**Issue by martell** [ninja-build/ninja#793]
Hey guys,
.
I'm trying to get ninja to build under mingw64 with a mingw64 compiled python from within msys2.

The issue is that if MSYSTEM is defined python even though it is a native windows exe will use '/' for paths instead of '\' because it knows that it is running from within a msys shell.

The result is

[25/26] AR build/libninja.a
FAILED: cmd /c ar cqs build/libninja.a.tmp build/build.o build/build_log.o build/clean.o build/debug_flags.o build/depfile_parser.o build/deps_log.o build/disk_interface.o build/edit_distance.o build/eval_env.o build/graph.o build/graphviz.o build/lexer.o build/line_printer.o build/manifest_parser.o build/metrics.o build/state.o build/util.o build/version.o build/subprocess-win32.o build/includes_normalize-win32.o build/msvc_helper-win32.o build/msvc_helper_main-win32.o build/getopt.o && move /Y build/libninja.a.tmp build/libninja.a
The system cannot find the path specified.
[25/26] CXX build/ninja.o
ninja: build stopped: subcommand failed.

Mingw-python is a native windows exe just like cpython but it just has extra checks to support the posix layer.

Would it be possible to get ninja to check windows for an environment variable MSYSTEM at runtime and if it is set then use mv instead of move.

The result would be a windows native windows exe that works just as normal but can detect if it is in a posix layer and change to suit.

please look here for more info
https://github.com/Alexpux/MINGW-packages/blob/master/mingw-w64-ninja/PKGBUILD
basically I want to remove CPython as a dependancy and use mingw-python instead from within the package manager.

Can someone tell me if this small change is possible?

**Comment by martell** [comment]
This is not an issue anymore :)

**Comment by stevenh** [comment]
Looks like this is broken again, build fails MSYS2 with:
```
The system cannot find the path specified.
bootstrapping ninja...
when running:  cmd /c ar cqs build/libninja.a.tmp build/build.o build/build_log.o build/clean.o build/clparser.o build/debug_flags.o build/depfile_parser.o build/deps_log.o build/disk_interface.o build/edit_distance.o build/eval_env.o build/graph.o build/graphviz.o build/lexer.o build/line_printer.o build/manifest_parser.o build/metrics.o build/state.o build/string_piece_util.o build/util.o build/version.o build/subprocess-win32.o build/includes_normalize-win32.o build/msvc_helper-win32.o build/msvc_helper_main-win32.o build/getopt.o && move /Y build/libninja.a.tmp build/libninja.a
Traceback (most recent call last):
  File "./configure.py", line 516, in <module>
    ninja_lib = n.build(built('libninja.a'), 'ar', objs)
  File "./configure.py", line 164, in build
    self._run_command(self._expand(cmd, local_vars))
  File "./configure.py", line 189, in _run_command
    subprocess.check_call(cmdline, shell=True)
  File "C:/msys64/mingw64/lib/python2.7/subprocess.py", line 187, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command 'cmd /c ar cqs build/libninja.a.tmp build/build.o build/build_log.o build/clean.o build/clparser.o build/debug_flags.o build/depfile_parser.o build/deps_log.o build/disk_interface.o build/edit_distance.o build/eval_env.o build/graph.o build/graphviz.o build/lexer.o build/line_printer.o build/manifest_parser.o build/metrics.o build/state.o build/string_piece_util.o build/util.o build/version.o build/subprocess-win32.o build/includes_normalize-win32.o build/msvc_helper-win32.o build/msvc_helper_main-win32.o build/getopt.o && move /Y build/libninja.a.tmp build/libninja.a' returned non-zero exit status 1
```
Running the following first from the linked script fixed it for me:
```shell
sed -i 's|cmd /c $ar cqs $out.tmp $in && move /Y $out.tmp $out|$ar crs $out $in|g' configure.pl
```
